src/Nancy/HttpLink.cs

```diff
@@ -0,0 +1,262 @@
+        /// </returns>
+        bool IEquatable<HttpLink>.Equals(HttpLink other)
+        {
+            if (ReferenceEquals(other, null))
+            {
+                return false;
+            }
+
+            if (this.TargetUri != other.TargetUri)
+            {
+                return false;
+            }
+
+            foreach (var parameter in other)
+            {
+                if (!this.ContainsKey(parameter.Key))
+                {
+                    return false;
+                }
+
+                var parameterValue = this[parameter.Key];
+
+                if (parameterValue == null || parameter.Value == null)
+                {
+                    continue;
+                }
+
+                if (!parameterValue.Equals(parameter.Value))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Determines whether the specified <see cref="System.Object" />, is equal to this instance.
+        /// </summary>
+        /// <param name="other">The <see cref="System.Object" /> to compare with this instance.</param>
+        /// <returns>
+        ///   <c>true</c> if the specified <see cref="System.Object" /> is equal to this instance; otherwise, <c>false</c>.
+        /// </returns>
+        public override bool Equals(object other)
+        {
+            return ((IEquatable<HttpLink>)this).Equals(other as HttpLink);
+        }
+
+        /// <summary>
+        /// Returns a hash code for this instance.
+        /// </summary>
+        /// <returns>
+        /// A hash code for this instance, suitable for use in hashing algorithms and data structures like a hash table.
+        /// </returns>
+        public override int GetHashCode()
+        {
+            return this.ToString().ToLowerInvariant().GetHashCode();
+        }
+
+        /// <summary>
+        /// Returns a <see cref="System.String"/> that represents this instance.
+        /// </summary>
+        /// <returns>
+        /// A <see cref="System.String"/> that represents this instance.
+        /// </returns>
+        public override string ToString()
+        {
+            var linkBuilder = new StringBuilder();
+            linkBuilder.Append('<');
+            linkBuilder.Append(this.TargetUri);
+            linkBuilder.Append('>');
+
+            foreach (var parameter in this.OrderBy(parameter => parameter.Key, new LinkParameterComparer()))
+            {
+                linkBuilder.Append("; ");
+                linkBuilder.Append(parameter.Key);
+                linkBuilder.Append('=');
+                linkBuilder.Append('"');
+                linkBuilder.Append(parameter.Value);
+                linkBuilder.Append('"');
+            }
+
+            return linkBuilder.ToString();
+        }
+
+        private T GetParamValue<T>(string paramName)
```

Comment:
Unless "param" is something that's specific to the RFC, I think we should rename to "parameter". We generally try to avoid that kind of shortening. More descriptive == better :smile: